subsys/logging/log_core.c

```diff
@@ -28,6 +28,161 @@ static bool initialized;
 
 static timestamp_get_t timestamp_func;
 
+/* MOJ KOD */
+
+#ifdef CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE
+	struct LOG_FRONTEND_PIN_NUMBERS log_frontend_pin_nums = {
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+1,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+2,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+3,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+4,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+5,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+6,
+		CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE_FIRST_BIT+7,
+		CONFIG_LOG_FRONTEND_PIN_CLK
+		};
+#else
+	struct LOG_FRONTEND_PIN_NUMBERS log_frontend_pin_nums = {
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_FIRST_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_SECOND_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_THIRD_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_FOURTH_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_FIFTH_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_SIXTH_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_SEVENTH_BIT,
+		CONFIG_LOG_FRONTEND_PINS_NOT_IN_SEQUENCE_EIGHTH_BIT,
+		CONFIG_LOG_FRONTEND_PIN_CLK
+		};
+#endif
+
+uint32_t mask_of_pins_log = 0x000005FE;
+
+uint32_t create_mask_gpio(struct LOG_FRONTEND_PIN_NUMBERS * pin_nums, uint8_t message)
+{
+	uint32_t mask = 0;
+
+	if(IS_ENABLED(CONFIG_LOG_FRONTEND_PINS_IN_SEQUENCE))
```

Comment:
In Zephyr conditional statements and loops require to have starting bracket in the same line.
`if (condition) {`